lib/events.js

```diff
@@ -437,9 +441,9 @@ EventEmitter.prototype.removeAllListeners =
         this.removeListener(type, listeners);
       } else if (listeners) {
         // LIFO order
-        do {
-          this.removeListener(type, listeners[listeners.length - 1]);
-        } while (listeners[0]);
+        for (i = listeners.length; i-- > 0;) {
```

Comment:
Looks like this is the only place in core we use this pattern (although it appears a few times in v8 c++ code. It was added in https://github.com/nodejs/node/commit/3e64b5677f10abf849c2da762421437b89d1e11e. I think we should remove this and just use the standard for loop style.

```
➜ rg "i[-|+]+ [<|>]" | grep for
lib/events.js:        for (i = list.length; i-- > 0;) {
deps/v8/src/deoptimizer.cc:    for (size_t i = count; i-- > 0;) {
deps/v8/src/ast/ast-types.cc:  for (size_t i = BoundariesSize() - 1; i-- > 0;) {
deps/v8/src/full-codegen/full-codegen.cc:    for (int i = last; i-- > 0;) {
deps/v8/src/compiler/types.cc:  for (size_t i = BoundariesSize() - 1; i-- > 0;) {
deps/v8/src/parsing/parser.cc:    for (int i = labels->length(); i-- > 0;) {
deps/openssl/openssl/crypto/store/str_lib.c:        for (i = 0; i++ < STORE_ATTR_TYPE_NUM;)
```